### ISSUE TITLE
False Warning: --honour-dir-layout (-j) makes no sense without --merge-directories (-D)

### ISSUE THREAD

**Issue by NoSuck** [sahib/rmlint#333]
Hello.  I just started using rmlint.  In fact, this is my very first rmlint command:

```
■ LC_ALL=C rmlint -Djfr -T df -c sh:symlink .
WARNING: --honour-dir-layout (-j) makes no sense without --merge-directories (-D)
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
ERROR: Failed to get parent path: stat failed: No such file or directory
...
```

Why the warning, and could the errors be related?  Clearly, I specified both `-D` and `-j`.  It is not a parsing issue, as separating arguments as `-D -j` (instead of `-Dj`) results in the same, confusing warning.

Thank you for reading.

```
■ rmlint --version
version 2.8.0 compiled: Dec 31 2018 at [18:06:43] "Maidenly Moose" (rev unknown)
compiled with: +mounts +nonstripped +fiemap +sha512 +bigfiles +intl +replay +xattr +btrfs-support

rmlint was written by Christopher <sahib> Pahl and Daniel <SeeSpotRun> Thomas.
The code at https://github.com/sahib/rmlint is licensed under the terms of the GPLv3.
■ uname -srvmo
Linux 4.20.0-arch1-1-ARCH #1 SMP PREEMPT Mon Dec 24 03:00:40 UTC 2018 x86_64 GNU/Linux
```

**Comment by NoSuck** [comment]
Note:  I might have expected a warning in this case regarding `-D` and no `-T dd` or other such type.  In fact, the above warning disappears after adding `dd` to the type list.  Is this related?

**Comment by sahib** [comment]
Hello @NoSuck, thanks for your report.

> Note: I might have expected a warning in this case regarding -D and no -T dd or other such type. In fact, the above warning disappears after adding dd to the type list. Is this related?

Indeed it is, specifying ``-T df`` effectively disables searching for duplicate directories (since you said *"search for duplicates only!"*). Internally ``rmlint`` just checks at a later point if the two options are given at the same time. I agree that the wording could be more fine grained here. I tried to make the message a little nicer in 3cdfc03898abc5363d01a558df805b1fa19d2a7e.

> Why the warning, and could the errors be related? Clearly, I specified both -D and -j. It is not a parsing issue, as separating arguments as -D -j (instead of -Dj) results in the same, confusing warning.

The errors are not related. It seems that the ``-f`` option does seem to cause those errors when used in conjunction with ``-T df``. So, actually you found a silly bug that crept in one of the last few releases. This bug happens when a symbolic link is followed (``-f``) and contains a relative path. The relative path is relative to the link itself, not to the directory ``rmlint`` is running in. Normally ``rmlint`` is handling absolute paths only, that's why he got confused. With the fix, he's building correct paths again. Please try it again with a ``rmlint`` version [compiled from develop](https://rmlint.readthedocs.io/en/latest/install.html#compilation).

**Comment by NoSuck** [comment]
Thank you for the response.  I tried 5ec79283de1600371cbf7b7ce7e2be2eff985615 and found that both warning and stat-error issues had been resolved.  That was quick!

I look forward to using rmlint for years to come.  Take care.
